chain/cosmos/src/trigger.rs

```diff
@@ -69,50 +69,7 @@ pub enum CosmosTrigger {
     Message(Arc<codec::MessageData>),
 }
 
-impl CheapClone for CosmosTrigger {
-    fn cheap_clone(&self) -> CosmosTrigger {
-        match self {
-            CosmosTrigger::Block(block) => CosmosTrigger::Block(block.cheap_clone()),
-            CosmosTrigger::Event { event_data, origin } => CosmosTrigger::Event {
-                event_data: event_data.cheap_clone(),
-                origin: *origin,
-            },
-            CosmosTrigger::Transaction(transaction_data) => {
-                CosmosTrigger::Transaction(transaction_data.cheap_clone())
-            }
-            CosmosTrigger::Message(message_data) => {
-                CosmosTrigger::Message(message_data.cheap_clone())
-            }
-        }
-    }
-}
-
-impl PartialEq for CosmosTrigger {
```

Comment:
do we have any coverage for this?

Reply:
I don't think we do. I'll think of a way to test this, even though it looks correct at a glance.

Reply:
you can prolly compare the expanded code

Reply:
I'll mark this as resolved as I have decided not to tackle it at this time, I don't feel like sweating over determinism issues. I've limited this PR to the `CheapClone` refactor.